src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubBuildStatusNotificationStrategy.java

```diff
@@ -0,0 +1,97 @@
+package org.jenkinsci.plugins.github_branch_source;
+
+import hudson.model.Job;
+import hudson.model.Result;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import jenkins.scm.api.*;
+import org.jenkinsci.plugins.displayurlapi.DisplayURLProvider;
+import org.kohsuke.github.GHCommitState;
+
+public class GitHubBuildStatusNotificationStrategy {
```

Comment:
Start with an `AbstractGitHubNotificationStrategy implements ExtensionPoint`  that just has `List<GitHubNotificationDetails> notifications(GitHubNotificationRequest, TaskListener)` then make this class be `DefaultGitHubNotificationStrategy extends AbstractGitHubNotificationStrategy`

Reply:
Return a `List` so that a strategy can decide to make multiple notifications or none... and the list avoids having to check for null ;-)

Reply:
Is there a way to make the Default subclass accessible by other extensions, so that they can defer to the default implementation for some parts of the notification? For example, I'm mostly interested in overriding the context label with an arbitrary string and my extension would be happy to keep everything else intact.

Reply:
As long as it’s a public class they can extend. But structuring this way allows for more flexibility later